tensor2tensor/data_generators/text_encoder.py

```diff
@@ -113,18 +124,18 @@ def __init__(self, vocab_filename, reverse=False, num_reserved_ids=2):
     self._reverse = reverse
     if vocab_filename is not None:
```

Comment:
here a vocab filename has to be provided so remove the line `if vocab_filename is not None:` and the `else` stanza.